s3compatapi/src/main/java/com/snowflake/s3compatapitestsuite/compatapi/S3CompatStorageClient.java

```diff
@@ -98,12 +100,27 @@ public String getBucketLocation(String bucketName) {
                 if (correctRegion != null) {
                     this.s3Client.setSignerRegionOverride(correctRegion);
                     this.s3Client.setRegion(RegionUtils.getRegion(correctRegion));
-                    return this.s3Client.getBucketLocation(bucketName);
+                    return this.s3Client.getBucketLocation(bucketName).toLowerCase();
                 }
             }
             throw ex;
         }
-        return regionRes;
+        return regionRes.toLowerCase();
+    }
+    public String getBucketRegionThroughMetadata(String bucketName) {
+        GetObjectMetadataRequest objectMetadataRequest = new GetObjectMetadataRequest(bucketName, "");
+        if (measurementPerformance && perfMeasurement != null) {
```

Comment:
do you wanna add try, catch? and for AmazonS3Exception try to see Region header in getAdditionalDetails? 
Most of the times this API response has region header present even on error.

Reply:
Sounds good. Try catch added.